src/mixings/dirichlet_mixing.cpp

```diff
@@ -0,0 +1,48 @@
+#include "dirichlet_mixing.hpp"
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <stan/math/prim/prob.hpp>
+
+#include "../../proto/cpp/mixings.pb.h"
+#include "../utils/rng.hpp"
+
+void DirichletMixing::update_hypers(
+    const std::vector<std::shared_ptr<BaseHierarchy>> &unique_values,
+    unsigned int n) {
+  if (state.has_fixed_value()) {
+    return;
+  } else if (state.has_gamma_prior()) {
+    auto rng = bayesmix::Rng::Instance().get();
```

Comment:
actually today I found a huge bug when working on a fork of this repo. We should change 
`auto rng = bayesmix::Rng::Instance().get();` with`auto& rng = bayesmix::Rng::Instance().get();`

Otherwise the internal state of the rng won't get changed

Reply:
Sure, will do in the master after merging